### PR TITLE
Service tests added

### PR DESCRIPTION
**Pull request by hpavlo**

### CODE REVIEW

**Review by SoulARC — APPROVED** [review]
Maybe it would be better to test the strategies separately, but I'm not sure it's necessary.. Overall great!

File: src/test/java/core/basesyntax/service/impl/DataParserImplTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import core.basesyntax.service.DataParser;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+
+public class DataParserImplTest {
+    private final DataParser dataParser = new DataParserImpl();
+
+    @Test
+    public void parseData_normalData_ok() {
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/service/impl/DataParserImplTest.java:15]: You also can test empty data

File: src/test/java/core/basesyntax/service/impl/DataReaderImplTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.service.DataReader;
+import java.util.List;
+import org.junit.Test;
+
+public class DataReaderImplTest {
+    private static final String NON_EXISTENT_FILE = "src/main/data.csv";
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/service/impl/DataReaderImplTest.java:10]: Also can check  reading  empty file

File: src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+
+public class FruitServiceImplTest {
+    private final FruitService fruitService = fruitServiceInitialise();
+
+    @Test
+    public void processData_normalFruitTransactions_ok() {
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java:24]: the same here You need to implement boundary value testing for your services (check it in other services).

File: src/test/java/core/basesyntax/service/impl/ReportWriterImplTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                + "banana,115" + LINE_SEPARATOR
+                + "apple,110" + LINE_SEPARATOR;
+
+        reportWriter.writeReport(report, NORMAL_FILE_PATH);
+        String actual;
+        try {
+            actual = Files.readString(Path.of(NORMAL_FILE_PATH));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file: " + NORMAL_FILE_PATH, e);
+        }
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/service/impl/ReportWriterImplTest.java:35]: better move it to a separate private method

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
good job! Let’s improve your solution 😉

**Pull request closed without merging**